.github/scripts/install-arduino-core-esp32.sh

```diff
@@ -27,6 +27,9 @@ if [ ! -d "$ARDUINO_ESP32_PATH" ]; then
     cd esp32
     #git submodule update --init --recursive > /dev/null 2>&1
 
+    echo "Copying CI restrictive warning settings from tools/platform.local.txt"
+    cp tools/platform.local.txt .
+    
```

Comment:
Wasn't this inside the `if else` statement above? It didn't work?
It should be added after line 20 (inside the `if`).

Reply:
it didn't work in side the `if`

Reply:
I had to move it outside in order to really copy the file

Reply:
I think it was inside the `else` not the `if` where it should be. Also there is a `cd` missing (or the path needs to be updated accordingly by just adding esp32/)

Reply:
You didn't force push, so we it's still here: https://github.com/espressif/arduino-esp32/pull/7060/commits/1cabb075155a6264f8a2d730c8c4cdc0a231988a

Reply:
I tested it inside `else` exactly as in https://github.com/espressif/arduino-esp32/commit/1cabb075155a6264f8a2d730c8c4cdc0a231988a
But it didn't work, CI was passing because `platform.local.txt` was not being copied.

After moving it ouside, it worked fine...

Reply:
In the CI the `else` path is not taken, this is why it wasn't being copied.  It's in the `if` where it should be.
`if [ ! -z "$GITHUB_REPOSITORY" ];  then`: if `GITHUB_REPOSITORY` is defined, i.e., this is a Github repo.

Reply:
If we copied it unconditionally, I think w will end up with a case of https://github.com/espressif/arduino-esp32/pull/7060#issuecomment-1204875043

Reply:
No, this is only copied in the CI docker space. It will never affect the end user, as suggested in https://github.com/espressif/arduino-esp32/pull/7060#issuecomment-1204875043

@igrr was actually referring to the first commit I made that added `platform.local.txt` to the main Arduino ESP32 Github folder, which would go to all end users and therefore affect them.

Reply:
I understand that the script can copy `tools/platform.local.txt` unconditionally, which is necessary for CI to consider desired Warning flags when compiling the examples.

Reply:
My assumption is that users won't run `.github/scripts/install-arduino-core-esp32.sh`, so -Werror is not going to affect them.

(Unless they want to replicate the behavior of CI; in which case they can run this script and should get the same result as in CI.)

Reply:
Test scripts do run `install-arduino-core-esp32.sh` to get some environment variables.  Running unit tests, as found under `tests` using `tests_build.sh` for instance will have this file copied.  Which is something users are supposed to do.
Other than this, this is a logic for Github, it should be under the Github path, regardless if it introduces issues or not.

Reply:
Ok, you are talking about HIL testing, right? Those PRs marked with HIL tag.
In that case it is still CI only, isn't it? 
So we will have -Werror=all and no -Wnoxxxx in HIL testing.
Do you see a problem with it?

Reply:
I was referring to running those tests locally using the `tests_build.sh` script.  It works similar to the CI.

Reply:
OK, thanks for clarifying. 
The absolute most of Arduino Users will never use `tests_build.sh`. 
Therefore, I'd say that if it happens, the user that does use it, is an expert one, and that person will know what to do.

I still see no problem with the script copying the file to the Docker file system.

Reply:
Let's take one step back please for me to understand the issue.  What's the problem with the below diff? It doesn't work?
```diff
diff --git a/.github/scripts/install-arduino-core-esp32.sh b/.github/scripts/install-arduino-core-esp32.sh
index 8584da5b..02f29c6a 100755
--- a/.github/scripts/install-arduino-core-esp32.sh
+++ b/.github/scripts/install-arduino-core-esp32.sh
@@ -18,6 +18,8 @@ if [ ! -d "$ARDUINO_ESP32_PATH" ]; then
     if [ ! -z "$GITHUB_REPOSITORY" ];  then
         echo "Linking Core..."
         ln -s $GITHUB_WORKSPACE esp32
+        echo "Copying CI restrictive warning settings from tools/platform.local.txt"
+        cp esp32/tools/platform.local.txt .
     else
         echo "Cloning Core Repository..."
         git clone https://github.com/espressif/arduino-esp32.git esp32 > /dev/null 2>&1
```

> I still see no problem with the script copying the file to the Docker file system.

The issues, as I see them, are:
1. This is a logic for Github but it's mixed with the common path and not with the Github one.
2. The script copies build altering files to the home directory unbeknownst to the user and also doesn't clean them.

As for who uses `tests_build.sh` I hope anyone writing drivers for Arduino would.  That means tests are written and validated locally.

Reply:
> Let's take one step back please for me to understand the issue. What's the problem with the below diff? It doesn't work?

It didn't work when it is executed by Github Action started by "On Push".
I see that when it uses GitHub-hosted runners with Ubuntu Image,  which is created fresh by the runner and it has nothing installed. So, I understand that the test `[ ! -z $GITHUB_REPOSITORY" ]` will always fail. 

>This is a logic for Github but it's mixed with the common path and not with the Github one.

This is just a Shell Script executed by a Runner that will use some VM with empty Ubuntu/Mac/Windows files system.
Then the script clones Arduino-IDE, ESP32 Arduino Core into it in order to excute the Tasks.
Right after cloning or linking (which I think never happens with Github Actions), it will copy `platform.local.txt` from `esp32/tools` to `esp32`. 

Looking into `install-arduino-core-esp32.sh` I can see other places that use Common Path - I just used it in the right place.
```
    if [ ! -z "$GITHUB_REPOSITORY" ];  then
        echo "Linking Core..."
        ln -s $GITHUB_WORKSPACE esp32
    else
        echo "Cloning Core Repository..."
        git clone https://github.com/espressif/arduino-esp32.git esp32 > /dev/null 2>&1
    fi

    #echo "Updating Submodules ..."
    cd esp32                                                   <<<<<<<<<<<<<<<<<<<<<<<
    #git submodule update --init --recursive > /dev/null 2>&1

    echo "Copying CI restrictive warning settings from tools/platform.local.txt"
    cp tools/platform.local.txt .
    
    echo "Installing Platform Tools ..."
    cd tools && python get.py    <<<<<<<<<<<<<<<<<<<<<<<<
    cd $script_init_path

    echo "ESP32 Arduino has been installed in '$ARDUINO_ESP32_PATH'"
    echo ""
```

> The script copies build altering files to the home directory unbeknownst to the user and also doesn't clean them.

Please clarify when does it happen. It copies files to the VM File System that only exists in the CI and Github Action Runner Time. Everything is destroyed after the end of VM execution. Nothing goes to the User File System, unless the script is executes directly in a Shell Command Line - which I understand is not the purpose of this script. 

### Anyway,  it works with Ubuntu/Mac/Windows and Arduino-Builder. It doesn't work with PlatformIO.

Reply:
> It didn't work when it is executed by Github Action started by "On Push".

Was this ever tried?

> Right after cloning or linking (which I think never happens with Github Actions),

You can check every on-push workflow, since the creation of this workflow, and you can see that it actually links the repo and does not clone it.  Cloning in Github Action is done with the Checkout Action.
<img width="389" alt="image" src="https://user-images.githubusercontent.com/12075908/183676158-e7984bb7-178d-435b-8edb-df6d230cd537.png">

<img width="903" alt="image" src="https://user-images.githubusercontent.com/12075908/183676239-34f3f0b4-b442-4cd7-a22d-0d15c196ceee.png">